internal/csi-addons/rbd/reclaimspace.go

```diff
@@ -69,6 +71,11 @@ func (rscs *ReclaimSpaceControllerServer) ControllerReclaimSpace(
 	defer rbdVol.Destroy()
 
 	err = rbdVol.Sparsify()
+	if errors.Is(err, rbdutil.ErrImageInUse) {
+		// treat sparsify call as no-op if volume is in use.
+		log.DebugLog(ctx, fmt.Sprintf("volume with ID %q is in use, skipping sparsify operation", volumeID))
+		return &rs.ControllerReclaimSpaceResponse{}, nil
```

Comment:
Ideally this returns `FAILED_PRECONDITION`, which should be added to https://github.com/csi-addons/spec/tree/main/reclaimspace#controllerreclaimspace-errors.

What is the behavior of kubernetes-csi-addons when this (now) undocumented error is returned?

Reply:
> Ideally this returns `FAILED_PRECONDITION`, which should be added to https://github.com/csi-addons/spec/tree/main/reclaimspace#controllerreclaimspace-errors.
> 
> What is the behavior of kubernetes-csi-addons when this (now) undocumented error is returned?


From a csidriver/storage perpective, we are free to choose to execute operation based on internal conditions too.
Since this is a significant issue which needs to be addressed immediately and this change can be backported, lets go ahead with this change.

We can discuss csiaddons level change further laters on(which will definitely not be backportable). 
That will involve a significant design chance in csiaddons spec, controller and csidriver.

Reply:
https://github.com/csi-addons/kubernetes-csi-addons/issues/406
I've opened an issue at k8s-csi-addons repo.
we can continue the discussion there.

Reply:
Thanks for creating the issue, that helps with the background information of the problem. As there it no documented way to abort/skip the operation, leave a `FIXME:` comment pointing to the related csi-addons/spec or csi-addons/kubernetes-csi-addons issue.